### PR TITLE
fix: fix remove note when board is locked

### PR DESCRIPTION
**Pull request by mateo-ivc**
## Description

User can delete own notes although the board is locked
## Changelog

In 'api/notes.go': Added check before deleting a note to check whether the board is locked or not
## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] The light- and dark-theme are both supported and tested
- [ ] The design was implemented and is responsive for all devices and screen sizes
- [x] The application was tested in the most commonly used browsers (e.g. Chrome, Firefox, Safari)

### CODE REVIEW

**Comment by mateo-ivc** [comment]
I would wait until the frontend is finished. Maybe some additional changes need to be made

File: server/src/api/context.go [hunk 1]
```diff
@@ -44,21 +44,55 @@ func (s *Server) BoardParticipantContext(next http.Handler) http.Handler {
...
 		log := logger.FromRequest(r)
 
+		var board uuid.UUID
 		boardParam := chi.URLParam(r, "id")
-		board, err := uuid.Parse(boardParam)
-		if err != nil {
-			common.Throw(w, r, common.BadRequestError(errors.New("invalid board id")))
+		if boardParam != "" {
+			var err error
+			board, err = uuid.Parse(boardParam)
```
Comment (brandstetterm) [inline server/src/api/context.go:51]:
```suggestion
			board, err := uuid.Parse(boardParam)
```
Doesn't that work?

File: server/src/api/context.go [hunk 2]
```diff
@@ -44,21 +44,55 @@ func (s *Server) BoardParticipantContext(next http.Handler) http.Handler {
...
+			if err != nil {
+				common.Throw(w, r, common.BadRequestError(errors.New("invalid board id")))
+				return
+			}
+		} else {
+			// board id is not defined in the url but maybe in the context
```
Comment (brandstetterm) [inline server/src/api/context.go:57]: Is there any endpoints where that's the case?

File: server/src/api/context.go [hunk 3]
```diff
@@ -44,21 +44,55 @@ func (s *Server) BoardParticipantContext(next http.Handler) http.Handler {
...
+			boardValue := r.Context().Value("Board")
+			log.Debugw("board id not found in url, using context", "board", boardValue)
+			if boardValue == nil {
+				common.Throw(w, r, common.BadRequestError(errors.New("board id not found")))
+				return
+			}
+			var ok bool
+			board, ok = boardValue.(uuid.UUID)
```
Comment (brandstetterm) [inline server/src/api/context.go:65]:
```suggestion
			board, ok := boardValue.(uuid.UUID)
```
Doesn't that work?

File: server/src/api/context.go (hunk 2 above)
Comment (mateo-ivc) [reply server/src/api/context.go:57]: I agree, that is an unnecessary check

**Comment by mateo-ivc** [comment]
Fixed issues with https://github.com/inovex/scrumlr.io/pull/4005

**Pull request closed without merging**